addons/website_sale/views/product_views.xml

```diff
@@ -98,8 +98,8 @@
                             <field name="website_id" options="{'no_create': True}" groups="website.group_multi_website"/>
                             <field name="website_sequence" groups="base.group_no_one"/>
                             <field name="public_categ_ids" widget="many2many_tags" string="Categories"/>
-                            <field name="alternative_product_ids" widget="many2many_tags" domain="[('id', '!=', active_id)]"/>
-                            <field name="accessory_product_ids" widget="many2many_tags"/>
+                            <field name="alternative_product_ids" widget="many2many_tags" domain="[('id', '!=', active_id), '|', ('company_id', '=', company_id), ('company_id', '=', False)]"/>
+                            <field name="accessory_product_ids" widget="many2many_tags" domain="[('product_tmpl_id', '!=', active_id), '|', ('company_id', '=', company_id), ('company_id', '=', False)]"/>
```

Comment:
The `product.product_template_form_view` is extended by `product.product_normal_form_view` and `product.product_template_only_form_view`.  Thus, `active_id` could be the id of a product.product but also of a product.template.

Suppose you have a _product.template_ 1 with 2 variants (1,2) and a _product.template_ 2 with variants (3-4-5).
If you access this form view for the product.product 2, you won't be able to choose a `product.product` whose template has id 2 :crying_cat_face: (e.g products 3, 4, 5) .

Not sure there is an easy way to fix this, but maybe the field is_product_variant could be useful (true for product.product and false for product.template) to show the field with a different domain depending on it.

Cc: @JKE-be

Reply:
`is_product_variant` is a non-stored field. So, can't be usable in a domain

Reply:
Not in the domain indeed, but you could maybe have the field defined two times in the view, with different domains.
* The first field definition is invisible when is_product_variant is True
* The second one is invisible when is_product_variant is False

I'm not sure it works perfectly fine, but if it works, it could be a good solution.

Reply:
@Feyensv  Yes! It's a good solution. But, here we have a limitation that we can't put the same field twice with a different domain in a view. A dynamic domain is not working. Thus, we have moved those fields with a specific domain in a separate view from the common view.
Is it ok for you?

Reply:
The solution works, and I don't see another way to do it :+1: .
But I'm not sure this "UI" change can land in stable, with the two views.
To see with @JKE-be ;).